src/algebra/homology/homotopy_category.lean

```diff
@@ -0,0 +1,224 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.homology.homotopy
+import category_theory.quotient
+
+/-!
+# The homotopy category
+
+`homotopy_category V c` gives the category of chain complexes of shape `c` in `V`,
+with chain maps identified when they are homotopic.
+-/
+
+universes v u
+
+open_locale classical
+noncomputable theory
+
+open category_theory category_theory.limits homological_complex
+
+variables {ι : Type*}
+variables (V : Type u) [category.{v} V] [has_zero_object V] [preadditive V]
+variables (c : complex_shape ι)
+
+/--
+`homotopy_category V c` is the category of chain complexes of shape `c` in `V`,
+with chain maps identified when they are homotopic.
+-/
+@[derive category]
+def homotopy_category :=
+category_theory.quotient (λ (C D : homological_complex V c) (f g : C ⟶ D), nonempty (homotopy f g))
+
+-- TODO the homotopy_category is preadditive
+
+namespace homotopy_category
+
+/-- The quotient functor from complexes to the homotopy category. -/
+def quotient : homological_complex V c ⥤ homotopy_category V c :=
+category_theory.quotient.functor _
+
+variables {V c}
+
+@[simp] lemma quotient_obj_as (C : homological_complex V c) :
+  ((quotient V c).obj C).as = C := rfl
+
+@[simp] lemma quotient_map_out {C D : homotopy_category V c} (f : C ⟶ D) :
+  (quotient V c).map f.out = f :=
+quot.out_eq _
+
+lemma eq_of_homotopy {C D : homological_complex V c} (f g : C ⟶ D) (h : homotopy f g) :
+  (quotient V c).map f = (quotient V c).map g :=
+category_theory.quotient.sound _ ⟨h⟩
+
+/-- We package up the fact that homotopy is closed under composition. -/
+lemma comp_closure_homotopy :
+  (quotient.comp_closure (λ (C D : homological_complex V c) (f g : C ⟶ D), nonempty (homotopy f g)))
+    = λ C D f g, nonempty (homotopy f g) :=
+begin
+  ext C D f g,
+  fsplit,
+  { rintro ⟨E, F, f, m₁, m₂, g, ⟨i⟩⟩,
+    split,
+    apply (i.comp_right _).comp_left, },
+  { intro h,
+    convert quotient.comp_closure.intro (𝟙 C) f g (𝟙 D) (by convert h);
+    simp, }
+end
+
+/-- We package up the fact that homotopy is an equivalence relation. -/
+lemma eqv_gen_homotopy {C D : homological_complex V c} :
+  (eqv_gen (λ (f g : C ⟶ D), nonempty (homotopy f g)))
+    = λ f g, nonempty (homotopy f g) :=
+begin
+  ext f g,
+  fsplit,
+  { intro h,
+    induction h with f g i h f g w i f g h w w' k l,
+    { exact i, },
+    { split, refl, },
+    { split, symmetry, exact i.some, },
+    { split, exact k.some.trans l.some, } },
+  { intro h,
+    exact eqv_gen.rel _ _ h, }
+end
+
+/-- If two chain maps become equal in the homotopy category, then they are homotopic. -/
+def homotopy_of_eq {C D : homological_complex V c} (f g : C ⟶ D)
+  (w : (quotient V c).map f = (quotient V c).map g) : homotopy f g :=
```

Comment:
It might be nice to have the general statement "if `r` is an equivalence and a congruence, then equality in `category_theory.quotient r` is equivalent to `r`"? Something along these lines was requested already in #2710.

Reply:
Yes, this would be good. I think we should add a `congruence` predicate in `category_theory.quotient`, and then write this lemma in terms of that.

(I will probably be slow making "extra feature" changes to this series of PRs, but if someone hasn't already done this by the time this reaches the top of the queue, I can have a go. If you want to do it first, please do. :-)

Reply:
#7501